### DISCUSSION TITLE
Wix custom action behavior gets invoked inconsistently on different windows server

### DISCUSSION THREAD

**Discussion by niha31-github** [wixtoolset/issues#6940] [Questions]
I have an issue where I have some custom action defined in wix project. We are creating patch for our product. To create patch, when we build the code to generate msi on win server 2008 then upon installing patch the custom action doesn't get invoked(expected behavior) but when we build the same code to generate msi on win server 2019 then the custom action is getting invoked.

The custom action is defined to update some ini file. Hence ini file is getting modified upon installing the patch.(which should not happen)

I have analyzed the MSIs created on both servers and could see the differences. Please find the diff 👍 

![image](https://user-images.githubusercontent.com/114738923/193253159-d39f9278-373e-414f-9e23-f1d8aac3be37.png)

In win server 2008 we are using Wix toolset v3.11.0.1528 and 2019 has 3.11.2.4516

I debug the patch log, found this Action  
WriteIniValues. Writing INI files values Executing op:IniFilePath(File=myfile.ini,Folder=C:\mypath) Executing op:IniWriteRemoveValue(Section=Data,Key=Server,Value=myserver,Mode=0) WriteIniValues: File: myfile.ini, Section: Data, Key: Server, Value: myserver Executing op:IniWriteRemoveValue(Section=Data,Key=Databasename,,Mode=0) WriteIniValues: File: myfile.ini, Section: Data, Key: Databasename, Value: Executing op: IniWriteRemoveValue(Section=Data,Key=Username,,Mode=0) WriteIniValues: File: myfile.ini, Section: Data, Key: Username, Value: 

See here, the ini file values are getting removed by writing empty values. These actions are not being invoked when i install the patch which was created from msi built on win 2008 server.

**Comment by robmen** [comment]
Seems related to [the question I just answered.](https://github.com/wixtoolset/issues/discussions/6953)
